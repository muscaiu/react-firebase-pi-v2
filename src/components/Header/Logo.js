import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import axios from 'axios';

import logo from "assets/img/logo.svg";

const rotate360 = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const LogoImg = styled.img`
  animation: ${props => props.isActive ? `${rotate360} infinite 20s linear` : 'null'};
  height: 150px;
`;

class Logo extends Component {
  static propTypes = {
    isActive: PropTypes.bool,
    showNotification: PropTypes.func
  };

  handleTestClick = () => {
    const { showNotification } = this.props;

    axios.get('http://cassusa.go.ro:3001/api/test')
      .then(function (response) {
        showNotification('bc', 'success', `Real status: ${response.data.relayStatus}`)
      })
      .catch(function (err) {
        showNotification('bc', 'danger', `${err}`)
      })
  }

  render() {
    const { isActive } = this.props;
    return (
      <LogoImg
        onClick={this.handleTestClick}
        isActive={isActive}
        src={logo}
      />
    )
  }
}

export default Logo;
