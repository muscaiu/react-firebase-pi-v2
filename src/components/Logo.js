import React from 'react';
import styled, { keyframes } from 'styled-components';
import axios from 'axios';

import logo from "assets/img/logo.svg";

const Wrapper = styled.div`
`;

const rotate360 = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const LogoImg = styled.img`
  animation: ${props => props.isActive ? `${rotate360} infinite 20s linear` : 'null'};
  height: 150px;
`;

const handleTestClick = () => {
  axios.get('http://cassusa.go.ro:3001/api/test')
    .then(function (response) {
      console.log(response.data.relayStatus)
      alert('Real status: ' + response.data.relayStatus);
    })
    .catch(function (err) {
      alert(err);
    })
}

const Logo = ({ isActive }) => (
  <Wrapper>
    <LogoImg
      onClick={handleTestClick}
      isActive={isActive}
      src={logo}
    />
  </Wrapper>
)

export default Logo;
