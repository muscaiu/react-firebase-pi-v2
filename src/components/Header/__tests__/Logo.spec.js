import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import createStore from 'redux-mock-store';

import Logo from '../Logo';
import logo from "assets/img/logo.svg";

describe('Logo', () => {
    const wrapper = shallow(<Logo />);

    it('should match snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('should contain a LogoImg', () => {
        // const state = {};
        // const mockStore = createStore();
        // const store = mockStore(state);
        // const wrapper = shallow(<Logo />, { context: { store } });
        // expect(wrapper.find('img')).toHaveLength(1);
        // expect(wrapper.find("img").prop("src")).toEqual(logo)
        console.log(wrapper.debug());
    });
})
