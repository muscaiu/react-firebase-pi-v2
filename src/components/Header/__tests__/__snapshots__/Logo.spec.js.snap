// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should match snapshot 1`] = `
<styled.img
  onClick={[Function]}
  src="logo.svg"
/>
`;
